src/common/buffer.hpp

```diff
@@ -284,104 +284,85 @@ class RxCounters {
     }
 };
 
-class Frame_stats {
+/**
+//  * This class stores the metrics coresponding to a Frame.
+//  * Specifically, it contains a) the number of symbols per frame and b) the number of tasks per symbol, per frame.
+//  */
+class FrameCounters {
 public:
     size_t max_symbol_count;
-    bool last_symbol(int frame_id)
-    {
-        const size_t frame_slot = frame_id % TASK_BUFFER_FRAME_NUM;
-        if (++symbol_count[frame_slot] == max_symbol_count) {
-            symbol_count[frame_slot] = 0;
-            return true;
-        }
-        return false;
-    }
+    size_t max_task_count;
 
-    void init(int _max_symbol_count)
+    void init(
+        int max_symbol_count, int max_task_count = -1, int max_data_symbol = -1)
     {
+        this->max_symbol_count = max_symbol_count;
+        this->max_task_count = max_task_count;
         symbol_count.fill(0);
-        max_symbol_count = _max_symbol_count;
+        //define task counters if #tasks is > 0
+        if (max_task_count > 0) {
+            if (max_data_symbol <= 0)
+                throw "the max data symbol should be greater than zero for a "
+                      "positive max task count.";
```

Comment:
We should use `rt_assert()` instead to be consistent with other error messages.
```suggestion
             rt_assert(max_data_symbol > 0, 
                 "The max data symbol should be larger than 0 for max_task_count > 0."); 
`````